src/crypto/ossl/cert.rs

```diff
@@ -25,16 +26,26 @@ pub struct OsslDtlsCert {
 
 impl OsslDtlsCert {
     /// Creates a new (self signed) DTLS certificate.
-    pub fn new() -> Self {
-        Self::self_signed().expect("create dtls cert")
+    pub fn new(options: DtlsCertOptions) -> Self {
+        Self::self_signed(options).expect("create dtls cert")
     }
 
     // The libWebRTC code we try to match is at:
     // https://webrtc.googlesource.com/src/+/1568f1b1330f94494197696fe235094e6293b258/rtc_base/openssl_certificate.cc#58
-    fn self_signed() -> Result<Self, CryptoError> {
+    fn self_signed(options: DtlsCertOptions) -> Result<Self, CryptoError> {
         let f4 = BigNum::from_u32(RSA_F4).unwrap();
-        let key = Rsa::generate_with_e(2048, &f4)?;
-        let pkey = PKey::from_rsa(key)?;
+        let pkey = match options.pkey_type {
+            DtlsPKeyType::Rsa => {
+                let key = Rsa::generate_with_e(2048, &f4)?;
+                PKey::from_rsa(key)?
+            }
+            DtlsPKeyType::EcDsa => {
+                let nid = Nid::X9_62_PRIME256V1; // NIST P-256 curve
```

Comment:
Is it understood that this is the curve to always use for ECDSA in this context?

Reply:
My understanding isn't great, but that seems to be the curve I've seen for 256bit EcDsa.

We could rename the enum values to EcDsaP256. and RSA to Rsa2048? To be ready when someone was EcDsaP384 or something else.

Reply:
Is there any precedence in Pion or similar?

Reply:
Looks like Pion doesn't allow you to select the key type.. it just uses EC-DSA w/ the NIST P-256 curve. https://github.com/pion/dtls/blob/1e4ae60e28516c475e999b389bcbe96a168c4470/pkg/crypto/selfsign/selfsign.go#L26

Reply:
you can use other curves but this is the common curve I've seen used. This is one of the FIPS standard curves, so that likely drives the adoption over others of similar strength.

Reply:
So I do think it would probably be good to rename the values in the enum to Rsa2048 and EcDsaP256 .. so you explicitly know the key size and curve being used.

And then it's simply to add other curves and strength in the future.